applications/TrilinosApplication/custom_python/add_trilinos_linear_solvers_to_python.cpp

```diff
@@ -69,7 +70,28 @@ void  AddLinearSolvers(pybind11::module& m)
 {
     py::class_<TrilinosLinearSolverType, TrilinosLinearSolverType::Pointer > (m,"TrilinosLinearSolver")
         .def(py::init<>())
-        .def("Solve", Solve);
+        .def("Solve", Solve)
+        ;
+
+    using TrilinosFallbackLinearSolverType = FallbackLinearSolver<TrilinosSparseSpaceType, TrilinosLocalSpaceType>;
+    py::class_<TrilinosFallbackLinearSolverType, TrilinosFallbackLinearSolverType::Pointer, TrilinosLinearSolverType>(m, "TrilinosFallbackLinearSolver")
+        .def(py::init<Parameters>())
+        .def(py::init<TrilinosLinearSolverType::Pointer, TrilinosLinearSolverType::Pointer, Parameters>())
+        .def(py::init<const std::vector<TrilinosLinearSolverType::Pointer>&, Parameters>())
+        .def("AddSolver", [](TrilinosFallbackLinearSolverType& rSelf, TrilinosLinearSolverType::Pointer pSolver) {
+            rSelf.AddSolver(pSolver);
+        })
+        .def("AddSolver", [](TrilinosFallbackLinearSolverType& rSelf, const Parameters ThisParameters) {
+            rSelf.AddSolver(ThisParameters);
+        })
+        .def("GetSolvers", &TrilinosFallbackLinearSolverType::GetSolvers)
+        .def("SetSolvers", &TrilinosFallbackLinearSolverType::SetSolvers)
+        .def("GetResetSolverEachTry", &TrilinosFallbackLinearSolverType::GetResetSolverEachTry)
+        .def("SetResetSolverIndexEachTry", &TrilinosFallbackLinearSolverType::SetResetSolverIndexEachTry)
+        .def("GetParameters", &TrilinosFallbackLinearSolverType::GetParameters)
+        .def("GetCurrentSolverIndex", &TrilinosFallbackLinearSolverType::GetCurrentSolverIndex)
+        .def("ClearCurrentSolverIndex", &TrilinosFallbackLinearSolverType::ClearCurrentSolverIndex)
+        ;
```

Comment:
(if we leave those functions after the discussion) you can refactor this into a function so that updates to the core will also be done here

Reply:
Okay